### PR TITLE
ISPN-14834 Implement CRC16 for hash function option

### PR DESCRIPTION
**Pull request by jabolina**
https://issues.redhat.com/browse/ISPN-14834

### CODE REVIEW

**Comment by jabolina** [comment]
Updated.

File: core/src/main/resources/schema/infinispan-config-15.0.xsd [hunk 1]
```diff
@@ -1702,6 +1712,16 @@
             </xs:documentation>
           </xs:annotation>
         </xs:attribute>
+        <xs:attribute name="hash-function" type="xs:string">
```
Comment (jabolina) [inline core/src/main/resources/schema/infinispan-config-15.0.xsd:1715]: I did not add an entry for scattered here.

**Comment by wburns** [comment]
Looks like we are missing an assertion in UnifiedXmlFileParsingTest to verify the hash function was applied properly.

**Comment by jabolina** [comment]
Ooops, added a verification.

**Comment by wburns** [comment]
Integrated into main, thanks @jabolina !

**Comment by pruivo** [comment]
I really don't like to have a new attribute that is only used when `key-partitioner=HashFunctionPartitioner`. IMO, we should have a new `KeyPartitioner` implementation for the new hash function: `class CRC16HashFunctionPartitioner extend HashFunctionPartitioner`.

WDYT @jabolina @wburns ?

**Comment by pruivo** [comment]
I agree, but it does not invalidate the fact that we have too many configuration attributes and this one is not required :+1:

**Comment by jabolina** [comment]
TBH, personally, I like having the `CRC16HashFunctionPartitioner`, it simplifies a bit of this code and the configuration side, too.

**Comment by wburns** [comment]
Sure to remove a config attribute it seems fine to me. I was more thinking of code side only regarding coupling and testability.

**Pull request merged**